common/src/main/java/com/wynntils/models/players/PartyModel.java

```diff
@@ -0,0 +1,404 @@
+ */
+public final class PartyModel extends Model {
+    // region Party Regexes
+    /*
+    Regexes should be named with this format:
+    PARTY_ACTION_INVOLVED_(DETAIL)
+    where:
+    PARTY should be the first word in the name
+    ACTION should be something like JOIN, LEAVE, LIST, etc.
+    INVOLVED should be the player that is affected by the action:
+    - ALL should be used if the action affects all players in the party
+    - LEADER should be used if the action affects the party leader
+    - SELF should be used if the action affects the player
+    - OTHER should be used if the action affects another player(s), but not ALL players
+    DETAIL (optional) should be a descriptor if necessary
+    */
+    private static final Pattern PARTY_LIST_ALL = Pattern.compile("§eParty members: (.*)");
+    private static final Pattern PARTY_LIST_LEADER = Pattern.compile("§r§b(.+)");
+    private static final Pattern PARTY_LIST_SELF_FAILED = Pattern.compile("§eYou must be in a party to list\\.");
+
+    private static final Pattern PARTY_LEAVE_OTHER = Pattern.compile("§e(.+) has left the party\\.");
+    private static final Pattern PARTY_LEAVE_SELF_ALREADYLEFT = Pattern.compile("§eYou must be in a party to leave\\.");
+
+    // This is a special case; Wynn sends the same message for when we leave a party or get kicked from a party
+    private static final Pattern PARTY_LEAVE_SELF_KICK = Pattern.compile("§eYou have been removed from the party\\.");
+
+    private static final Pattern PARTY_JOIN_OTHER = Pattern.compile("§e(.+) has joined the party\\.");
+    private static final Pattern PARTY_JOIN_OTHER_SWITCH =
+            Pattern.compile("§eSay hello to (.+) which just joined your party!");
+    private static final Pattern PARTY_JOIN_SELF = Pattern.compile("§eYou have successfully joined the party\\.");
+
+    private static final Pattern PARTY_PROMOTE_OTHER = Pattern.compile("§eSuccessfully promoted (.+) to party leader!");
+    private static final Pattern PARTY_PROMOTE_SELF =
+            Pattern.compile("§eYou are now the leader of this party! Type /party for a list of commands\\.");
+
+    private static final Pattern PARTY_DISBAND_ALL = Pattern.compile("§eYour party has been disbanded\\.");
+    private static final Pattern PARTY_DISBAND_SELF =
+            Pattern.compile("§eYour party has been disbanded since you were the only member remaining\\.");
+
+    private static final Pattern PARTY_CREATE_SELF = Pattern.compile("§eYou have successfully created a party\\.");
+
+    private static final Pattern PARTY_KICK_OTHER = Pattern.compile("§eYou have kicked the player from the party.\\.");
+    // endregion
+
+    private boolean expectingPartyMessage = false; // Whether the client is expecting a response from "/party list"
+    private boolean nextKickHandled = false; // Whether the next "/party kick" sent by the client is being handled
+
+    private boolean partying; // Whether the player is in a party
```

Comment:
🍰 Party like there's no tomorrow! :-) I don't know if it is my non-native English skills that play me a trick here, but "partying" seems to me to fully indicate the "festivities" meaning of "party", not the "association" meaning.

Maybe `inParty` or `partyActive` or, I dunno...